### PR TITLE
Run integration tests in parallel

### PR DESCRIPTION
**Pull request by mehulkar**
See https://github.com/Nicoretti/prysk/issues/165

```sh
$ hyperfine --warmup=3 --export-markdown=turbo-integration-tests.md 'make integration-tests' 'make integration-tests-parallel'

Benchmark 1: make integration-tests
  Time (mean ± σ):     55.926 s ±  1.367 s    [User: 52.293 s, System: 16.442 s]
  Range (min … max):   54.899 s … 59.114 s    10 runs
 
  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet PC without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.
 
Benchmark 2: make integration-tests-parallel
  Time (mean ± σ):     23.354 s ±  1.575 s    [User: 64.686 s, System: 25.129 s]
  Range (min … max):   21.640 s … 25.709 s    10 runs
 
Summary
  'make integration-tests-parallel' ran
    2.39 ± 0.17 times faster than 'make integration-tests'
```

```
$ cat turbo-integration-tests.md
```

| Command | Mean [s] | Min [s] | Max [s] | Relative |
|:---|---:|---:|---:|---:|
| `make integration-tests` | 55.926 ± 1.367 | 54.899 | 59.114 | 2.39 ± 0.17 |
| `make integration-tests-parallel` | 23.354 ± 1.575 | 21.640 | 25.709 | 1.00 |

### CODE REVIEW

File: cli/Makefile [hunk 1]
```diff
@@ -307,15 +307,22 @@ CRAM_ENV := .cram_env
...
+	$(CRAM_ENV)/bin/pip3 install --quiet pytest "prysk[pytest-plugin]" pytest-xdist
+
 INTEGRATION_TEST_FILES = $(shell find integration_tests -name "*.t")
 
-integration-tests: $(CRAM_ENV)/bin/prysk turbo $(INTEGRATION_TEST_FILES) corepack turbo
+integration-tests-serial: $(CRAM_ENV)/bin/prysk turbo $(INTEGRATION_TEST_FILES) corepack turbo
 	$(CRAM_ENV)/bin/prysk --shell=`which bash` $(INTEGRATION_TEST_FILES)
 
+integration-tests: $(CRAM_ENV)/bin/prysk $(CRAM_ENV)/bin/pytest turbo $(INTEGRATION_TEST_FILES) corepack turbo
+	$(CRAM_ENV)/bin/pytest -n auto $(INTEGRATION_TEST_FILES)
```
Comment (chris-olszewski) [inline cli/Makefile:324]: Let's use `bash` as a shell instead of `sh`
```suggestion
	$(CRAM_ENV)/bin/pytest -n auto $(INTEGRATION_TEST_FILES) --prysk-shell=`which bash`
```

File: cli/integration_tests/turbo_version.t [hunk 2]
```diff
@@ -2,7 +2,9 @@ Setup
   $ . ${TESTDIR}/setup.sh
 
 Test version matches that of version.txt
-  $ diff <(head -n 1 ${VERSION}) <(${TURBO} --version)
+  $ FROM_FILE=$(head -n 1 ${VERSION})
+  $ FROM_CLI="${TURBO} --version"
+  $ bash -c 'diff <(${FROM_FILE}) <(${FROM_CLI})'
```
Comment (chris-olszewski) [inline cli/integration_tests/turbo_version.t:7]: We can revert these changes if we configure our tests to use bash

**Review by chris-olszewski — COMMENTED** [review]
Stuff is passing for me locally, but it looks like cwd is getting weird on CI

**Comment by mehulkar** [comment]
> Stuff is passing for me locally, but it looks like cwd is getting weird on CI

yeah 😢 , looks like we reference the name of the test dir in a lot of setup scripts and that breaks. I'm not sure why yet.

**Review by chris-olszewski — APPROVED** [review]
If we can get another green run, I think this is fine to merge

**Comment by nathanhammond** [comment]
@mehulkar the `nested.t` stuff is required to disambiguate between the fixtures; if we're doing this we need to move that information into the fixtures themselves.
